Shared/Extensions/JellyfinAPI/TaskTriggerType.swift

```diff
@@ -0,0 +1,57 @@
+//
+// Swiftfin is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2024 Jellyfin & Jellyfin Contributors
+//
+
+import Foundation
+
+public enum TaskTriggerType: String, Codable, CaseIterable {
+    case daily = "DailyTrigger"
+    case weekly = "WeeklyTrigger"
+    case interval = "IntervalTrigger"
+    case startup = "StartupTrigger"
+    case manual = "ManualTrigger"
```

Comment:
I saw this and implemented the server-side enums so in the future we get a strong type from the API, however I didn't find anything for `manual` or `systemEvent`. What are these used for?

Additionally, we could implement this in the API and have as a custom patch:
- https://github.com/jellyfin/jellyfin-sdk-swift/tree/main/Plugins/CreateAPI/PatchFiles

Reply:
I was just pulling the types that I was finding here: 
https://github.com/jellyfin/jellyfin-web/blob/708226763a0644c0cddad7740269ca557de1b5ce/src/controllers/dashboard/scheduledtasks/scheduledtask.js

That being said, we're doing all of the task triggers as non-Identifiable arrays and strings so it's all kind of weird to work with.

For our usage, I don't think SystemEventTrigger is used or at least it's not accessible from a user perspective? I think we can get away with just Daily, Weekly, Interval, Startup.

Is there a good/better way to go from String -> Enum? I tacked in the unknown as a catch all.

Reply:
Scratch that, I do think we only need 4 for this configuration. There is also a ManualTrigger and SystemEventTrigger but for this usage, we only need these:

https://github.com/jellyfin/jellyfin-web/blob/708226763a0644c0cddad7740269ca557de1b5ce/src/controllers/dashboard/scheduledtasks/scheduledtask.html#L32

Reply:
`SystemEventTrigger` used to exist but was removed a bit ago. The references in web are old because web is old.

- https://github.com/jellyfin/jellyfin/commit/f68038f2cff4ac94ac5afe97b5bfb5c9e172e6ff#diff-11f2a02a773748634ec46e848855f3b0a01b4504897819015854f8a2cc5790d5

Reply:
Shoot. I use Web as my reference for a lot of this...

For what I am doing on this PR, do you want me to keep working with this enum or should I hold off and we can do the SDK changes first? I'm game either way. TBH, I think I have some reworking to do using the formatter instead of strings so starting over wouldn't impact me too much.

Reply:
Using the web as reference is just fine, we just should also make sure values are valid. We can still have the enum here until we update to using the latest SDK version.

You are free to do what you like with starting over.